windows.js

```diff
@@ -153,3 +219,36 @@ export function addVCVARSEnv() {
   }
   return newPathEntries
 }
+
+// ssl files cause issues with non RI2 Rubies (<2.4) and ruby/ruby's CI from
+// build folder due to dll resolution
+function renameSystem32Dlls() {
```

Comment:
This seems quite risky, is there any other way to solve the issue?
Is it only for Ruby <= 2.3? It seems done for all cases here.
It seems to work fine before for <= 2.3, no?

Reply:
This is quite involved as to why it's needed.

To be clear, these **are not MSFT files**.  One of the oldest and most popular Windows OpenSSL implementations (built with Visual C) installs the OpenSSL dll's in `C:\Windows\System32`.  Normally, that folder is considered 'off-limits', and should be used only by MSFT.

I've had this in setup-ruby-pkgs for a long time, and it's never caused an issue.  It causes all sorts of WTF issues if it's not done...

Reply:
My concern here is someone might waste a lot of time if for some reason some part of their workflow would use those files and using setup-ruby has the very unexpected side effect to delete or rename files under `C:\Windows\System32` (which are not provided by Microsoft but still).
I remember some people were already confused due to the more extensive Path changes on Windows by this action, but this goes another level. If we keep it we should have a clear logged message mentioning what it and why it does it.

I think it's better to not do this. If those files are problematic then filing an issue at https://github.com/actions/virtual-environments seems the best way forward.

@larskanis What do you think? Having a 3rd opinion might be helpful here.

Reply:
Below is info from a repo I've got that spits out a bunch of data on Actions Images, taken from the windows-2022 image job.  All of the openssl.exe files have the dll's in their own directory.  The `C:\Windows\System32` files are added by the same installer that creates the files at `C:\Program Files\OpenSSL`.
```
——————————————————————————————————————————————————————————————————————— libcrypto-1_1-x64.dll
C:\mysql\bin\libcrypto-1_1-x64.dll
C:\Windows\System32\libcrypto-1_1-x64.dll
C:\Program Files\OpenSSL\bin\libcrypto-1_1-x64.dll
C:\Program Files\Git\mingw64\bin\libcrypto-1_1-x64.dll
c:\tools\php\libcrypto-1_1-x64.dll

——————————————————————————————————————————————————————————————————————— libssl-1_1-x64.dll
C:\mysql\bin\libssl-1_1-x64.dll
C:\Windows\System32\libssl-1_1-x64.dll
C:\Program Files\OpenSSL\bin\libssl-1_1-x64.dll
C:\Program Files\Git\mingw64\bin\libssl-1_1-x64.dll
c:\tools\php\libssl-1_1-x64.dll

——————————————————————————————————————————————————————————————————————— openssl info
OpenSSL 1.1.1  11 Sep 2018    C:\Program Files\OpenSSL\bin\openssl.exe
OpenSSL 1.1.1l  24 Aug 2021   C:\Program Files\Git\mingw64\bin\openssl.exe
OpenSSL 1.1.1l  24 Aug 2021   C:\Program Files\Git\usr\bin\openssl.exe
```

> My concern here is someone might waste a lot of time

I've worked with quite a few extension gems re Windows, Ruby master/head, and Actions.  Never come across a problem.  It is a confusing problem if Rubies that use `ENV['Path']` for their dll resolution and load them because they're in `C:\Windows\System32`...

Reply:
Could you do this only for RI1, so Ruby <= 2.3, and log a message in that case?
I think it's more reasonable to use workarounds like this for old versions, but for recent versions we should avoid any such hacks.

Reply:
Windows OpenSSL dll's have different names, depending on the version, the break happens between 1.0.2 and 1.1.0.  Updated for Ruby 2.4 and earlier, as some extconf files that work with OpenSSL look for the 1.1.0 names first.  Windows Ruby 2.4 uses 1.0.2.  See comment